### ISSUE TITLE
Finetuning 

### ISSUE THREAD

**Issue by Harry-KIT** [AakiraOtok/YOWOv3#4]
hi @AakiraOtok 
I appreciate your quick response.
I am interested in How to Fine-Tune That model on small custom Dataset? 
Have you ever thought about creating code snippet to do that?

**Comment by AakiraOtok** [comment]
Thank you for your appreciation.

Of course! My research is focused heavily on supporting the community with this model, so the instructions are something that I will improve in the near future. However, as I am the sole developer and maintainer of this project, it will require more time. Once the preparation steps are completed, I will provide comprehensive and specific guidance on finetuning the model, customizing datasets, as well as customizing components within the model. Please stay tuned for the release of the YOWOv3 paper. <3

**Comment by Harry-KIT** [comment]
Hi @AakiraOtok,

I trained the model using a custom 4-class dataset (falling, fighting, smoking, abandon) based on the provided configuration. However, I am experiencing an issue where the loss value doesn't decrease and remains around 1.3. Could you provide any feedback or suggestions on how to address this issue?

**Comment by AakiraOtok** [comment]
@Harry-KIT Hi,  
Did you fine-tune using a provided checkpoint or train from scratch?

**Comment by Harry-KIT** [comment]
Yes I did like you said using the provided weights.

**Comment by AakiraOtok** [comment]
In the main branch, I haven't updated for the fine-tuning task. I assume you have modified the code to handle the mismatch between the num_classes in the config file and the checkpoint. Please check this to ensure there are no bugs.

In my experiments, I observed that the loss decreases to around 2.7 - 3.4. Your loss seems slightly lower than what I have seen. The loss for predicting bounding boxes has been learned and is likely to change less, whereas the loss for predicting classes is something I think needs to be tracked. You can try increasing the scale_cls_loss to see if it actually reduces during the training process. Note that there are two loss functions, "simota" and "tal," so make sure you are modify the corresponeding one.

**Comment by Harry-KIT** [comment]
Thank you for detail guideline. It is training now. we may discuss it after the training is over. Could you give me your mail address for the further discussion if you don't mind?

**Comment by AakiraOtok** [comment]
Of course! I don't have a work email address, but you can contact me at the following address: [22520847@gm.uit.edu.vn](mailto:22520847@gm.uit.edu.vn)
